src/cluster_legacy.c

```diff
@@ -3767,8 +3767,8 @@ int clusterProcessPacket(clusterLink *link) {
                             /* This packet is stale so we avoid processing it anymore. Otherwise
                              * this may cause a primary-replica chain issue. */
                             return 1;
-                        } else if (nodeIsReplica(sender_claimed_primary)) {
-                            serverAssert(sender_claimed_primary->replicaof == sender);
+                        } else if (nodeIsReplica(sender_claimed_primary) &&
+                                   sender_claimed_primary->replicaof == sender) {
```

Comment:
This change looks safe and it avoids the crash that we have seen in CI. I think we need to merge it.

Reply:
Not that we got ACK from Ping, i am merging it.